internal/core/software_renderer.rs

```diff
@@ -40,6 +40,24 @@ type PhysicalPoint = euclid::Point2D<i16, PhysicalPx>;
 
 type DirtyRegion = PhysicalRect;
 
+/// This enum describes which parts of the buffer passed to the [`SoftwareRenderer`] may re-use to speed up painting.
+#[derive(PartialEq, Eq, Debug, Clone, Default)]
+pub enum RepaintBufferType {
+    #[default]
+    /// The full window is always redrawn. No attempt at tracking dirty items will be made.
```

Comment:
I suggest to avoid introducing a phrase here ("tracking dirty items").

```suggestion
    /// The full window is always redrawn. No attempt at partial rendering will be made.
```